File: pset6/caesar.py
import cs50
import sys

if len(sys.argv) < 2:
    print("no arguments")
    exit(1)

if len(sys.argv) > 2:
    print("error")
    exit(2)

key = int(sys.argv[1])
print("plaintext: ", end="")
str1 = cs50.get_string()
str2 = []
str1 = list(str1)
i = 0
while(i < len(str1)):
    if ord(str1[i]) >= 97 and ord(str1[i]) <= 122:
        tmp = ord(str1[i]) - 97
        tmp = tmp + key
        tmp = tmp % 26
        tmp = tmp + 97
        str1[i] = chr(tmp)

    elif ord(str1[i]) >= 65 and ord(str1[i]) <= 90:
        tmp = ord(str1[i]) - 65
        tmp = tmp + key
        tmp = tmp % 26
        tmp = tmp + 65
        str1[i] = chr(tmp)

    i = i + 1

str1 = ''.join(str1)
print("ciphertext: {}".format(str1))

File: pset2/caesar.c
#include<cs50.h>
#include<string.h>
#include<stdio.h>
#include<math.h>
int useratoi(string x)
{
    int key,i,j;
    for(j=0,i=strlen(x)-1;i>=0;i--,j++)
  key=key + pow(10,j)*(x[i]-48);
  
  return key;
}
int main(int y,string command[])

{
    if(y<2)
    {
    printf("no arguments\n");
     return 1;
    }
    if(y>2)
    {
        printf("error\n");
        return 2;
    }
    
    int key,i=0,tmp;
    string instr,str,str2;
    instr=malloc(strlen(command[1])+1);
    strcpy(instr,command[1]);
    key=useratoi(instr);//a user defined function to parse into int
    printf("plaintext: ");
    str=get_string();
    str2=malloc(strlen(str)+1);
    
    while(str[i]!='\0')
    {
       
        if(str[i]>=97&&str[i]<=122)
        
        {
            
       
        tmp=str[i]-97;
        tmp=tmp+key;
        tmp=tmp%26;
       tmp=tmp+97;
       str2[i]=tmp;
        
        }
     else  if(str[i]>=65&&str[i]<=90)
       {  
        
        tmp=str[i]-65;
        tmp=tmp+key;
        tmp=tmp%26;
        tmp=tmp+65;
       str2[i]=tmp;
        
        }
      else str2[i]=str[i];
      
      
     i++;
    
    }
 printf("ciphertext: %s\n",str2);          

    
    return 0;
}